From a8a4d23f21cf56ca5d23a2ba15981b1f6459f19b Mon Sep 17 00:00:00 2001
From: Allen Ye <allen.ye@mediatek.com>
Date: Wed, 30 Aug 2023 18:34:47 +0800
Subject: [PATCH 02/15] wifi: mt76: mt7915: check sta rx control frame to
 multibss capability

Add CAP3_RX_CTRL_FRAME_TO_MULTIBSS check when setting sta_muru_tlv,
which is used to get stations's capability of receving control frames
when running OFDMA with MBSSIDs.

Signed-off-by: Allen Ye <allen.ye@mediatek.com>
Signed-off-by: Shayne Chen <shayne.chen@mediatek.com>
---
 mt7915/mcu.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mt7915/mcu.c b/mt7915/mcu.c
index 6d000841..35cd4fb0 100644
--- a/mt7915/mcu.c
+++ b/mt7915/mcu.c
@@ -906,6 +906,8 @@ mt7915_mcu_sta_muru_tlv(struct mt7915_dev *dev, struct sk_buff *skb,
 		HE_MAC(CAP2_MU_CASCADING, elem->mac_cap_info[2]);
 	muru->ofdma_ul.uo_ra =
 		HE_MAC(CAP3_OFDMA_RA, elem->mac_cap_info[3]);
+	muru->ofdma_ul.rx_ctrl_frame_to_mbss =
+		HE_MAC(CAP3_RX_CTRL_FRAME_TO_MULTIBSS, elem->mac_cap_info[3]);
 }
 
 static void
-- 
2.39.2

